sklearn/multioutput.py

```diff
@@ -979,22 +1034,25 @@ class labels for each estimator in the chain.
         ],
     }
 
+    # TODO(1.8): Remove base_estimator from __init__
     def __init__(
         self,
-        base_estimator,
+        estimator=None,
```

Comment:
```suggestion
        estimator,
```

Reply:
if you pass `estimator` without default, it will become a required parameter. I thought the goal was only deprecate `base_estimator`. for example,
```
ClassifierChain(base_estimator=svc)
```
would not raise error but warning

Reply:
`base_estimator` was a positional argument. Thus we also want `estimator` to be.

In the past, a user could not write the following:

```python
ClassifierChain()
```

An error would be raised at instantiation. Now, a user will not get the error before calling `.fit`.

So I don't see the point to allow a estimator that can be constructed but will raise later an error.

When it comes to making the warning, the idea is to raise a warning when a user specifically passed:

```python
ClassifierChain(base_estimator=svc)
```

When we can detect easily this case by affecting `base_estimator="deprecated"` and check in `fit` that `base_esimator != "deprecated"`.

Reply:
> So I don't see the point to allow a estimator that can be constructed but will raise later an error.

I get what you mean and I can add if statement to check that `estimator` is not `none`.

But, it is possible that there exist some code out there that uses scikit-learn API to do machine learning, and it possible that they used the API like this:
```
ClassifierChain(base_estimator=svc)
```
Similar to #23819, i thought the goal is to only raise `FutureWarning` for backward compatibility.

Reply:
Oh I see, I overlooked this. So indeed, we will need `estimator` to be also a keyword to detect the `base_estimator=...` case. So one thing, that we can do, is to use the `Hidden` parameter validation option to avoid to expose it.

I'll make a comment to show how to do that.